### ISSUE TITLE
crashes using unnumbered lists

### ISSUE THREAD

**Issue by cptechnik** [stefan-niedermann/nextcloud-notes#335]
just make

test-notice{enter}
 - one{enter}
 - two{enter}
{close notice} 
{reopen}
{input two "enter" after text "two" ...CRASH}

**Comment by cptechnik** [comment]
Ich kann natürlich auch deutsch... ;)

**Comment by stefan-niedermann** [comment]
Reproducable with v0.13.1

**Comment by korelstar** [comment]
Yep, sounds like a bug in [RxMarkdown](https://github.com/yydcdut/RxMarkdown).

**Comment by unsyn** [comment]
Also crashing when leaving ordered lists as described.

**Comment by stefan-niedermann** [comment]
This is the corresponding stack trace if. @korelstar sure this is not caused by us? I can not find a single reference to RxMarkdown...

```
04-10 09:49:36.027 2758-2758/it.niedermann.owncloud.notes E/AndroidRuntime: FATAL EXCEPTION: main
    java.lang.RuntimeException: Unable to start activity ComponentInfo{it.niedermann.owncloud.notes/it.niedermann.owncloud.notes.android.activity.NotesListViewActivity}: android.view.InflateException: Binary XML file line #44: Error inflating class ImageView
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:1956)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:1981)
        at android.app.ActivityThread.access$600(ActivityThread.java:123)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1147)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:137)
        at android.app.ActivityThread.main(ActivityThread.java:4424)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:511)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551)
        at dalvik.system.NativeStart.main(Native Method)
     Caused by: android.view.InflateException: Binary XML file line #44: Error inflating class ImageView
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:697)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:739)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:742)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:742)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:742)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:742)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:352)
        at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:287)
        at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:139)
        at it.niedermann.owncloud.notes.android.activity.NotesListViewActivity.onCreate(NotesListViewActivity.java:109)
        at android.app.Activity.performCreate(Activity.java:4466)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1049)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:1920)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:1981) 
        at android.app.ActivityThread.access$600(ActivityThread.java:123) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1147) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:137) 
        at android.app.ActivityThread.main(ActivityThread.java:4424) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:511) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551) 
        at dalvik.system.NativeStart.main(Native Method) 
     Caused by: android.content.res.Resources$NotFoundException: Resource is not a Drawable (color or path): TypedValue{t=0x1/d=0x7f080078 a=-1 r=0x7f080078}
        at android.content.res.Resources.loadDrawable(Resources.java:1902)
        at android.content.res.TypedArray.getDrawable(TypedArray.java:601)
        at android.widget.ImageView.<init>(ImageView.java:119)
        at android.support.v7.widget.AppCompatImageView.<init>(AppCompatImageView.java:71)
        at android.support.v7.widget.AppCompatImageView.<init>(AppCompatImageView.java:67)
        at android.support.v7.app.AppCompatViewInflater.createImageView(AppCompatViewInflater.java:181)
        at android.support.v7.app.AppCompatViewInflater.createView(AppCompatViewInflater.java:105)
        at android.support.v7.app.AppCompatDelegateImplV9.createView(AppCompatDelegateImplV9.java:1035)
        at android.support.v7.app.AppCompatDelegateImplV9.onCreateView(AppCompatDelegateImplV9.java:1092)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:668)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:739) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:742) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:742) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:742) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:742) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:489) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:396) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:352) 
        at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:287) 
        at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:139) 
        at it.niedermann.owncloud.notes.android.activity.NotesListViewActivity.onCreate(NotesListViewActivity.java:109) 
        at android.app.Activity.performCreate(Activity.java:4466) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1049) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:1920) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:1981) 
        at android.app.ActivityThread.access$600(ActivityThread.java:123) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1147) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:137) 
        at android.app.ActivityThread.main(ActivityThread.java:4424) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:511) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551) 
        at dalvik.system.NativeStart.main(Native Method) 
```

**Comment by godfuture** [comment]
I have also noticed that the app crashes when I press enter to get next line / list item.

**Comment by KlausHuber** [comment]
I got this crash as well. Steps to reproduce are exactly as told by cptechnik.

My logcat entry looks like this (let me know if you need more details):

```
AndroidRuntime: Shutting down VM
AndroidRuntime: FATAL EXCEPTION: main
AndroidRuntime: Process: it.niedermann.owncloud.notes, PID: 29864
AndroidRuntime: java.lang.IndexOutOfBoundsException: getChars (86 ... 87) ends beyond length 84
AndroidRuntime: 	at android.text.SpannableStringBuilder.checkRange(SpannableStringBuilder.java:1265)
AndroidRuntime: 	at android.text.SpannableStringBuilder.getChars(SpannableStringBuilder.java:1147)
AndroidRuntime: 	at android.text.TextUtils.getChars(TextUtils.java:89)
AndroidRuntime: 	at android.text.SpannableStringBuilder.<init>(SpannableStringBuilder.java:66)
AndroidRuntime: 	at android.text.SpannableStringBuilder.subSequence(SpannableStringBuilder.java:1139)
AndroidRuntime: 	at com.yydcdut.rxmarkdown.live.CodeLive.onTextChanged(CodeLive.java:71)
AndroidRuntime: 	at com.yydcdut.rxmarkdown.live.LivePrepare.onTextChanged(LivePrepare.java:81)
AndroidRuntime: 	at com.yydcdut.rxmarkdown.RxMDEditText.onTextChanged4Controller(RxMDEditText.java:253)
AndroidRuntime: 	at com.yydcdut.rxmarkdown.RxMDEditText.access$700(RxMDEditText.java:45)
AndroidRuntime: 	at com.yydcdut.rxmarkdown.RxMDEditText$EditTextWatcher.onTextChanged(RxMDEditText.java:177)
AndroidRuntime: 	at android.widget.TextView.sendOnTextChanged(TextView.java:8195)
AndroidRuntime: 	at android.widget.TextView.handleTextChanged(TextView.java:8257)
AndroidRuntime: 	at android.widget.TextView$ChangeWatcher.onTextChanged(TextView.java:10383)
AndroidRuntime: 	at android.text.SpannableStringBuilder.sendTextChanged(SpannableStringBuilder.java:1208)
AndroidRuntime: 	at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:578)
AndroidRuntime: 	at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:509)
AndroidRuntime: 	at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:508)
AndroidRuntime: 	at android.view.inputmethod.BaseInputConnection.replaceText(BaseInputConnection.java:844)
AndroidRuntime: 	at android.view.inputmethod.BaseInputConnection.commitText(BaseInputConnection.java:198)
AndroidRuntime: 	at com.android.internal.widget.EditableInputConnection.commitText(EditableInputConnection.java:183)
AndroidRuntime: 	at com.android.internal.view.IInputConnectionWrapper.executeMessage(IInputConnectionWrapper.java:353)
AndroidRuntime: 	at com.android.internal.view.IInputConnectionWrapper$MyHandler.handleMessage(IInputConnectionWrapper.java:93)
AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
AndroidRuntime: 	at android.os.Looper.loop(Looper.java:241)
AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6274)
AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
ActivityManager: New dropbox entry: it.niedermann.owncloud.notes, data_app_crash, 345ea87a-0708-4e59-a035-0fa8d282a39e
ActivityManager:   Force finishing activity it.niedermann.owncloud.notes/.android.activity.EditNoteActivity
```

**Comment by dtcg** [comment]
Same crash confirmed on Android 7 device.

**Comment by natrius** [comment]
Reproduceable.
Android 8.1, LineageOS 15.1
OnePlus 3
App Version 0.17.0 (via F-Droid)

**Comment by stefan-niedermann** [comment]
This should be fixed with #422 which has been resolved. Is anyone able to still reproduce this behaviour with the current master? (The new version has not been released yet)

**Comment by natrius** [comment]
Nope, updated to 0.20.0 still crashing.

**Comment by cptechnik** [comment]
0.17.0 don't crash... hmm...

**Comment by stefan-niedermann** [comment]
@natrius yeah, the bug is only fixed in master branch. The fix will be shipped with v0.21.0.
